### PR TITLE
Make us do all testing only once

### PR DESCRIPTION
**Pull request by ChadNedzlek**
Exclude the categories we use.

Also, since we run "dotnet test", don't have the build also do the test,
since those results aren't uploaded, it's 100% impossible to see what even failed.

### CODE REVIEW

File: eng/test.yaml [hunk 1]
```diff
@@ -1,44 +1,12 @@
...
-  condition: succeededOrFailed()
-
-- task: Powershell@2
-  inputs: 
-    targetType: 'filePath'
-    filePath: eng\convert-codecoveragetoxml.ps1
-    arguments: -Path "$(system.defaultworkingdirectory)" -NugetPackagesPath "$(Build.SourcesDirectory)\.packages"
-  displayName: Convert Code Coverage to XML (powershell)
-
-- task: Palmmedia.reportgenerator.reportgenerator-build-release-task.reportgenerator@4
```
Comment (missymessa) [inline eng/test.yaml:27]: Do you know if by refactoring this to use VSTest if it will actually upload the code coverage reports as they are today?

**Comment by missymessa** [comment]
So it looks like by removing all those tasks to build up the code coverage report, it doesn't create the report in a nice readable way in AzDO: https://dev.azure.com/dnceng/public/_build/results?buildId=657885&view=codecoverage-tab 

You have to download the report and look at it in Visual Studio, which was something I was avoiding by setting it up like this in the first place. This also breaks the consistency in reporting that I documented [here](https://github.com/dotnet/core-eng/tree/master/Documentation/Validation#consistency-in-reporting) since the rest of the repos that we have configured this way report code coverage in the nice reports displayed on AzDO without extra steps to download the report and view through Visual Studio. 

I understand and agree with only running the tests once, but I think you're taking a step backwards by removing the tasks that create nicely formatted reports that are immediately viewable on AzDO.

File: eng/test.yaml (hunk 1 above)
Comment (missymessa) [reply eng/test.yaml:27]: These steps should be replaced. You can refactor whatever you need to in order to run the tests, but please ensure that the functionality to create and upload the formatted reports remain.

**Comment by ChadNedzlek** [comment]
Indeed, this went bad, so I need to go back to the drawing board. :-)

File: eng/test.yaml (hunk 1 above)
Comment (ChadNedzlek) [reply eng/test.yaml]: Totally revamped the PR. Let's see how the new one goes.

**Comment by ChadNedzlek** [comment]
Alrighty, I think I might have it this time!

**Comment by ChadNedzlek** [comment]
It worked this time! the Null ref is gone!

**Pull request merged**